packages/components/src/lib/utility/platform.js

```diff
@@ -32,16 +32,36 @@ export default class Platform {
       this.constructor.evt.ORIENTATION_CHANGE.pub({ orientation: this.orientation })
     }
 
+    /*
+    If document is in a quirks mode (see https://www.w3.org/TR/2016/WD-cssom-view-1-20160317/#dom-element-clientheight)
+    then `document.body.clientHeight` should be used instead of `document.documentElement.clientHeight`.
+    In a quirks mode a value of `document.compatMode` will be "BackCompat"
+    (see https://developer.mozilla.org/en-US/docs/Web/API/Document/compatMode)
+     */
+    const clientHeight = (document.compatMode === 'BackCompat')
+      ? document.body.clientHeight
+      : document.documentElement.clientHeight
+
     this.viewport = {
       width: window.innerWidth && document.documentElement.clientWidth && document.body.clientWidth
         ? Math.min(window.innerWidth, document.documentElement.clientWidth, document.body.clientWidth)
         : document.body.clientWidth || window.innerWidth || document.documentElement.clientWidth,
 
+      // The following should calculate a viewport height correctly in both quirks and non-quirks modes
```

Comment:
but you're still only using documentElement.clientHeight here? why is the clientHeight const only used for the scrollbars?

Reply:
This works, but is not very precise. However, extreme precision was not important for our purpose so I decided not to change it because it is a tested code that is used in several places.

`window.innerHeight` works correctly in both modes. This is a height of a viewport including a scrollbar.

In a "no-quirks" mode `documentElement.clientHeight` is a height of a viewport without a scrollbar. In a "quirks" mode is it a height of a document (a value usually higher than a viewport width). So, depending on the mode and a scrollbar presence `Math.min(window.innerHeight, document.documentElement.clientHeight)` will return a height of a viewport with or without a scrollbar. It is not a problem for the popup height as we have extra margins of 60px and, because scrollbar height is 17px, it will be an extra gap of 17px sometimes, almost not noticeable. It will appear only when a popup is too tall and its height is restricted.

But since I'm touching this probably it will be good to make adjustments here too. Maybe we can have a `viewport`  (or maybe call it outerViewport) to report dimensions including a scrollbar (i.e. innerWidth/innerHeight), then have a scrollbar width props, and then a getters that will return width and height of of a viewport without sidebars (maybe we can call it `innerViewport`? Not sure what's the best name for it)?

What do you think?

Reply:
I think that's a good idea. Because I think otherwise we could come back to this at some point and forget why we only applied it in one place. inner and outer viewport sound like ok names to me. thanks.

Reply:
Great, will update it tomorrow.